x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/support/Variables.java

```diff
@@ -22,7 +22,13 @@
     public static final String METADATA = "metadata";
     public static final String VARS = "vars";
 
-    public static Map<String, Object> createCtxModel(WatchExecutionContext ctx, Payload payload) {
+    public static Map<String, Object> createCtxParamsMap(WatchExecutionContext ctx, Payload payload) {
+        Map<String, Object> model = new HashMap<>();
+        model.put(CTX, createCtx(ctx, payload));
+        return model;
+    }
+
+    public static Map<String, Object> createCtx(WatchExecutionContext ctx, Payload payload) {
```

Comment:
do we still need this? should it be @deprecated?

Reply:
Not sure what you mean. This was extracted from the previous method to allow only getting the actual ctx map, but not putting it into an outer params map (the new contexts need this so they can put it into the wrapped ParameterMap for deprecation warnings).

Reply:
yea i just mean if we dont want coders (me and @spinscale) accidentally using the createCtx method, u could private it, or if other things are still using it, mark it @deprecated in favor of using the other method `createCtxParamsMap`

Reply:
Both are used for now. I will add a comment to try to make it clearer which should be used when.

Reply:
could you have just a second `createCtxParamsMap` method that takes another map as parameter? Which is an empty hash map by default used to fill and the parmaters in `WatcherConditionScript`?

Reply:
I think having 2 methods with the same name would be more confusing. Do the java docs I added not make sense?